app/src/main/java/com/androidimpact/app/activities/RecipeAddEditIngredientActivity.java

```diff
@@ -44,34 +59,90 @@ protected void onCreate(Bundle savedInstanceState) {
+            returnIntent.putExtra("ingredient", ingredient);
+            setResult(Activity.RESULT_OK, returnIntent);
+            finish();
+        }
+    }
 
-        // OnClickListeners on the page
-        Button addButton = findViewById(R.id.add_button);
-        addButton.setOnClickListener(v -> {
-            if (description.getText().toString().isBlank() || amount.getText().toString().isBlank() || unit.getText().toString().isBlank() || category.getText().toString().isBlank()) {
-                Toast toast = Toast.makeText(getApplicationContext(), "Blank input!", Toast.LENGTH_LONG);
-                toast.show();
-            } else {
-                Ingredient ingredient = new Ingredient(description.getText().toString(), Float.parseFloat(amount.getText().toString()), unit.getText().toString(), category.getText().toString());
-                Log.i(TAG + ":confirmed", "Added ingredient");
-                Intent returnIntent = new Intent();
-                returnIntent.putExtra("ingredient", ingredient);
-                setResult(Activity.RESULT_OK, returnIntent);
-                finish();
+    /**
+     * This method handles clicks on the cancel button
+     * @param view
+     *    The view that this method is connected to (cancel button)
+     */
+    public void cancel(View view) {
+        Log.i(TAG + ":cancel", "Cancel ingredient add");
+        setResult(Activity.RESULT_CANCELED);
+        finish();
+    }
+
+    /**
+     * This method checks if a recipe can be added with the values in the fields.
+     * @return
+     *   returns true if values are fine, false otherwise.
+     */
+    public boolean checkInputs() {
+
+        // Code adapted from groupmate Aneeljyot Alagh in his Assignment 1
+        // Accessed on October 30, 2022
+        String[] blankCheckStrings = {"Description", "Amount", "Unit", "Category"}; // mandatory fill out
+        ArrayList<String> toastMessage = new ArrayList<>();
+        boolean invalidInput = false;
+        boolean[] blankChecks = {
+                getStr(description).isBlank(),
+                getStr(amount).isBlank(),
+                getStr(unit).isBlank(),
+                getStr(category).isBlank()
+        };
+
+        // Make sure all inputs are filled
+        for (int i = 0; i < blankChecks.length; i++) {
+            if (blankChecks[i]) {
+                invalidInput = true;
+                toastMessage.add(blankCheckStrings[i]);
             }
-        });
+        }
 
-        Button cancelButton = findViewById(R.id.cancel_button);
-        cancelButton.setOnClickListener(v -> {
-            Log.i(TAG + ":cancel", "Cancel ingredient add");
-            setResult(Activity.RESULT_CANCELED);
-            finish();
-        });
+        if (invalidInput){
+            // If blanks, only print blank messages
+            Toast toast = Toast.makeText(getApplicationContext(), String.join(", ", toastMessage) + " must be filled!", Toast.LENGTH_LONG);
```

Comment:
Can you change this toast to a snackbar? Snackbars should be a bit more consistent across the system, and has a better UX.

https://stackoverflow.com/a/34432504

Reply:
Hey Josh, thanks for sending this! I agree snackbars are better as they get dismissed when switching activities.